scripts/reboot_smartswitch_helper

```diff
@@ -0,0 +1,293 @@
+    local message=$1
+    echo "$(date '+%Y-%m-%d %H:%M:%S') - $message" >&2
+}
+
+# Function to check if running on smart switch
+function is_smartswitch()
+{
+    python3 -c "from utilities_common.chassis import is_smartswitch; print(is_smartswitch())" | grep -q "True"
+}
+
+# Function to check if running on DPU
+function is_dpu()
+{
+    python3 -c "from utilities_common.chassis import is_dpu; print(is_dpu())" | grep -q "True"
+}
+
+# Function to retrieve number of DPUs
+function get_num_dpus()
+{
+    python3 -c "from utilities_common.chassis import get_num_dpus; print(get_num_dpus())"
+}
+
+# Function to retrieve DPU IP from CONFIG_DB
+function get_dpu_ip()
+{
+    local DPU_NAME=$1
+    sonic-db-cli CONFIG_DB HGET "DHCP_SERVER_IPV4_PORT|bridge-midplane|${DPU_NAME}" "ips@"
+}
+
+# Function to retrieve GNMI port from CONFIG_DB
+function get_gnmi_port()
+{
+    local DPU_NAME=$1
+    sonic-db-cli CONFIG_DB HGET "DPU_PORT|$DPU_NAME" "gnmi"
+}
+
+# Function to get reboot status from DPU
+function get_reboot_status()
+{
+    local dpu_ip=$1
+    local port=$2
+    local reboot_status
+    reboot_status=$(docker exec -i gnmi gnoi_client -target "${dpu_ip}:${port}" -logtostderr -insecure -rpc RebootStatus 2>/dev/null)
+    if [ $? -ne 0 ] || [ -z "$reboot_status" ]; then
+        log_message "Error: Failed to send reboot status command to DPU ${DPU_NAME}"
+        return ${EXIT_ERROR}
+    fi
+    local is_reboot_active
+    is_reboot_active=$(echo "$reboot_status" | grep "active" | awk '{print $2}')
+    if [ "$is_reboot_active" == "false" ]; then
+        log_message "DPU ${DPU_NAME} has finished rebooting"
+        return ${EXIT_SUCCESS}
+    fi
+    return ${EXIT_ERROR}
+}
+
+# Function to detach PCI module
+function pci_detach_module()
+{
+    local DPU_NAME=$1
+    local DPU_BUS_INFO=$2
+    python3 -c "from reboot_helper import pci_detach_module; pci_detach_module('${DPU_NAME}')"
+    if [ $? -ne 0 ]; then
+        log_message "Error: PCI detach vendor API is not available"
+        echo 1 > /sys/bus/pci/devices/${DPU_BUS_INFO}/remove
+    fi
+}
+
+# Function to rescan PCI module
+function pci_reattach_module()
+{
+    local DPU_NAME=$1
+    local DPU_BUS_INFO=$2
+    python3 -c "from reboot_helper import pci_reattach_module; pci_reattach_module('${DPU_NAME}')"
+    if [ $? -ne 0 ]; then
+        log_message "Error: PCI reattach vendor API is not available"
+        echo 1 > /sys/bus/pci/rescan
+    fi
+}
+
+# Function to reboot DPU
+function reboot_dpu_platform()
+{
+    local DPU_NAME=$1
+    local REBOOT_TYPE=$2
+    python3 -c "import reboot_helper; reboot_helper.reboot_dpu('${DPU_NAME}', '${REBOOT_TYPE}')"
+}
+
+# Function to wait for DPU reboot status
+function wait_for_dpu_reboot_status()
+{
+    local dpu_ip=$1
+    local port=$2
+
+    if [[ -z "$PLATFORM_JSON_PATH" ]]; then
+        log_message "Error: PLATFORM_JSON_PATH is not defined"
+        exit $EXIT_ERROR
+    fi
+
+    local dpu_halt_services_timeou
```

Comment:
dpu_halt_services_timeout?

Reply:
it would be better to merge line 108 and 109:
```
local dpu_halt_services_timeout=$(jq -r '.dpu_halt_services_timeout' "$PLATFORM_JSON_PATH" 2>/dev/null)
```

Reply:
Fixed it. Thank you!